### PR TITLE
feat: seven-verb CLI, MCP tools, and gem facades

### PR DESCRIPTION
**Pull request by gildesmarais**
## What changed

- Seven curation verbs end-to-end: `inspect`, `recon`, `capture`, `validate`, `test`, `apply`, `scrape` across CLI, gem facades (`lib/html2rss.rb`), and MCP (`Server::Tools` registry)
- New domain services: `Recon` (verdict BUILD/DEFER/DROP + native feed), `Test` (schema + live extraction + `FailureKind`), `Batch` (bounded thread-pool concurrency)
- MCP refactor: data-driven tool registration, `Outcome::Playbook` next_step SSOT, `Runtime` wire ownership; tool renames (`batch_scrape`, `batch_inspect`, `batch_recon`, bare verb names)
- `PageRecon::Diagnostics` moved from MCP namespace; `CLI::Validate` extracted; `CLI::Render` / `ProbeView` for output
- Historic CLI aliases preserved: `feed` → `apply`, `auto` → `scrape`
- **Breaking:** `Html2rss.capture` returns `Capture::CaptureResult` (use `.yaml` / `.config`)
- Review residuals: validate path heuristic, MCP test strategy respect, recon discovery error notes, golden-path docs (`capture → test → apply`)

## Why

Contributors and curation agents needed a single composable lifecycle instead of fragmented CLI commands, ad-hoc batch scripts, and MCP tools with stale `_url`/`_config` suffixes. One wire contract reduces round-trips and aligns CLI, gem API, and MCP playbooks.

## Risk

- **Breaking Ruby API:** `Html2rss.capture` return type; deleted `batch_auto_feed`, `Batch.inspect_urls` / `Batch.scrape_urls` (see CHANGELOG migration table)
- **Breaking MCP wire:** tool IDs renamed to bare verbs; external MCP clients must re-register
- Batch operations spawn worker threads (max concurrency 10 per batch)
- Downstream repos (`html2rss-configs`, `html2rss-web`, Cursor MCP namespace) need coordinated bumps

## Review map

1. `lib/html2rss/mcp/server/tools.rb` + `lib/html2rss/mcp/outcome/playbook.rb` — tool registration, next_step routing, inspect ≠ recon boundary
2. `lib/html2rss/recon.rb` + `lib/html2rss/test.rb` + `lib/html2rss/batch.rb` — domain services and failure modes
3. `lib/html2rss/cli.rb` + `lib/html2rss/cli/validate.rb` — CLI surface and review-fix edge cases
4. `lib/html2rss.rb` — gem facades and breaking `CaptureResult` API
5. `spec/integration/curation_golden_path_spec.rb` — Outcome policy routing across verbs

## Validation

- `make ready` exit 0 (1787 examples, 0 failures; RuboCop, YARD, schema sync, SimpleCov 98.41% line)

### CODE REVIEW

File: lib/html2rss/capture.rb [hunk 1]
```diff
@@ -65,30 +117,91 @@ def initialize(url, strategy: :auto, **options)
...
+      strategy = FeedPipeline::StrategyPlan.concrete_for_diagnostic(@strategy)
+      RequestSession.build(
+        config:,
+        strategy:,
+        budget: resources.budget,
+        policy: resources.policy
+      )
+    end
+
+    def infer_topics(text) # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity, Metrics/AbcSize, Metrics/MethodLength
```
Comment (gildesmarais) [inline lib/html2rss/capture.rb]: Stop the growth, remove
